### DISCUSSION TITLE
example `httpserverbmp` require a screen?

### DISCUSSION THREAD

**Discussion by stc1988** [Moddable-OpenSource/moddable#1445] [General]
I try to run example  `httpserverbmp`  with m`5stamp_s3` which has no screen, but I saw below error while build application.

```
Error: no screen configured
 at setup/commodetto ()
### Error: no screen configured!
```

I understand that Commodetto is required to generate BMP, but I am not sure if it is correct for such a situation to occur when nothing is actually displayed on the screen.
I haven’t tried it, but I think the same can be said about the `camera-server`.

**Comment by phoddie** [comment]
Thanks for the report.

A little while ago, I changed the `setup/commodetto` [implementation  to throw](https://github.com/Moddable-OpenSource/moddable/blob/public/build/devices/esp32/setup/commodetto.js#L25) that warning when there is no screen. That is helpful in some cases but, unfortunately, it breaks the `httpserverbmp` example. Previously, it would silently run.

The following code should work for you. It [replaces lines 21 to 28](https://github.com/Moddable-OpenSource/moddable/blob/9198d0111edab82bf68bd4e4747add00df0c3039/build/devices/esp32/setup/commodetto.js#L21-L28). The error is now a warning traced to the console.

```js
import config from "mc/config";

if (!config.Screen)
	trace("WARNING: no screen configured\n");

export default function (done) {
	if (!global.screen && config.Screen) {
		globalThis.screen = new config.Screen({});
```

**Reply by stc1988** [reply]
Thank you for confirming. This code works as expected!!

**Reply by phoddie** [reply]
Perfect. Thanks for the quick confirmation. I'll get that applied to the next release.
